test/e2e/minimal_configuration.go

```diff
@@ -22,14 +22,15 @@ package e2e
 import (
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	operatorv1 "sigs.k8s.io/cluster-api-operator/api/v1alpha1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-var _ = Describe("Create providers with minimal specified configuration", func() {
-	It("should succefully create a CoreProvider", func() {
+var _ = Describe("Create, upgrade, downgrade and delete providers with minimal specified configuration", func() {
+	It("should successfully create a CoreProvider", func() {
```

Comment:
Do we want to add a delete test for CoreProvider as well?

Reply:
Yes, I wrote a test for that (line 396). The thing is that in order to install other providers, we need Core Provider to be Ready. It means we can't delete it immediately and have to postpone this action.

Reply:
Ah right, was thinking the same initially, thanks for clarification

Reply:
@Fedosin feel free to _resolve_ the conversation